src/sprout/Controllers/AdminController.php

```diff
@@ -807,6 +807,189 @@ public function importAction($type)
+    **/
+    public function emailReportAdd($type)
+    {
+        AdminAuth::checkLogin();
+
+        $view = new PhpView('sprout/admin/main_layout');
+        $this->setDefaultMainviewParams($view);
+
+        $ctlr = $this->getController($type);
+        if (! $ctlr) return;
+        if (! $this->checkAccess($ctlr, 'email_report', false)) return;
+
+        $this->setNavigation($view, $ctlr);
+
+        $main = $ctlr->_addEmailReport();
+        if ($main instanceof AdminError) {
+            $this->error($main->getMessage(), $ctlr);
+            return;
+        }
+
+        if (! is_array($main)) $main = array('title' => $ctlr->getFriendlyName(), 'content' => $main);
+
+        $view->browser_title = strip_tags($main['title']);
+        $view->main_title = $main['title'];
+        $view->main_content = $main['content'];
+
+        unset($_SESSION['admin']['field_values']);
+        unset($_SESSION['admin']['field_errors']);
+
+        echo $view->render();
+    }
+
+    /**
+    * Executes the save action for a specific email report
+    *
+    * @param string $type The type of item to save
+    **/
+    public function emailReportAction($type)
+    {
+        AdminAuth::checkLogin();
+        Csrf::checkOrDie();
+
+        /** @var ManagedAdminController */
+        $ctlr = Admin::getController($type);
+        if (! $ctlr) return;
+        if (! $this->checkAccess($ctlr, 'email_report', true)) return;
+
+        $_SESSION['admin']['field_values'] = $_POST;
+
+        $valid = new Validator($_POST);
+        $valid->required(['email_report_name', 'email_report_format', 'multiedit_recipients']);
+
+        $has_email = false;
+        foreach($_POST['multiedit_recipients'] as $recipient) {
+            if (empty($recipient['name']) and empty($recipient['email'])) continue;
+            $has_email = true;
+            if (empty($recipient['name']) or empty($recipient['email'])) {
+                $valid->addGeneralError('Recipients must contain both a name and email');
+                Notification::error('Recipients must contain both a name and email');
+            }
+        }
+
+        if (!$has_email) {
+            $valid->addGeneralError('Please add at least one recipient');
+            Notification::error('Please add at least one recipient');
+        }
+
+        if ($valid->hasErrors()) {
+            $_SESSION['admin']['field_errors'] = $valid->getFieldErrors();
+            Notification::error('Please ensure all required fields are completed');
+            Url::redirect('admin/email_report_add/' . $ctlr->getControllerName());
+        }
+
+        Pdb::transact();
+        $now = Pdb::now();
+
+        $filters = $_POST;
+        unset($filters['edit_token']);
+        unset($filters['multiedit_recipients']);
+        foreach ($filters as $filter_name => $filter_val) {
+            if (strpos($filter_name, 'email_report_') !== false) {
+                unset($filters[$filter_name]);
+            }
+        }
```

Comment:
What's the story here? Can filters not be be in a nested array?

Reply:
I changed this to just pass through the filters already json encoded - and prompt when creating a report without updating changes